### PR TITLE
build: Add auto-approve action for dependabot

### PR DESCRIPTION
**Pull request by ju-nc**
https://github.com/tophat/commit-utils/issues/46

Note: this shouldn't be merged until some test coverage is added

PR adding tests: https://github.com/tophat/commit-utils/pull/51

### CODE REVIEW

**Comment by noahnu** [comment]
You're going to want to create a dependabot config and move off of the preview: https://github.com/tophat/syrupy/blob/master/.github/dependabot.yml. This does mean we'll lose auto-merge without a bit of work though.

File: .github/workflows/auto-approve.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Auto approve
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: hmarr/auto-approve-action@v2.0.0
+      if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
```
Comment (ju-nc) [inline .github/workflows/auto-approve.yml:9]: dependabot-preview check is probably redundant here

**Review by noahnu — DISMISSED** [review]
We should make sure our test coverage is sufficient to be using dependabot

File: .github/workflows/auto-approve.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: hmarr/auto-approve-action@v2.0.0
+      if: github.actor == 'dependabot[bot]'
+      with:
+        github-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (noahnu) [inline .github/workflows/auto-approve.yml:11]: You need to add a GITHUB token to the project's secrets. I'll generate one and add it under GH_TOKEN (for consistency). Can you update this name?

Comment (ju-nc) [reply .github/workflows/auto-approve.yml]: Done!

**Review by noahnu — APPROVED** [review]
LGTM, although this won't do anything b/c there's no auto merge.

As another thought, I wonder if we could change the if condition so it waits for all other status checks to be successful

**Pull request merged**